cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/DRepActivity.hs

```diff
@@ -0,0 +1,301 @@
+hprop_check_drep_activity :: Property
+hprop_check_drep_activity = H.integrationWorkspace "test-activity" $ \tempAbsBasePath' -> do
+    -- Start a local test net
+  conf@Conf { tempAbsPath } <- mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoNodeEra = cEra
+        , cardanoNumDReps = 10
+        }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets=wallet0:wallet1:wallet2:_
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  epochStateView <- getEpochStateView (File configurationFile) (File socketPath)
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  -- First proposal (set activity to something feasible in the test)
+  dRepActivityBeforeFirstProp <- getDRepActivityValue execConfig
+  dRepActivityBeforeFirstProp === 100 -- This is the default value
+
+  let firstPropVotes :: [(String, Int)]
+      firstPropVotes = zip (concatMap (uncurry replicate) [(6, "yes"), (4, "no")]) [1..]
+  annotateShow firstPropVotes
+
+  fgaInfo@(firstGovernanceActionTxId, firstGovernanceActionIndex) <-
+    makeActivityChangeProposal execConfig epochStateView (File configurationFile) (File socketPath)
+                               sbe gov "proposal1" Nothing 4 wallet0
+
+  voteChangeProposal execConfig epochStateView sbe gov "vote1"
+                     firstGovernanceActionTxId firstGovernanceActionIndex firstPropVotes wallet0
+
+  (EpochNo epochAfterFirstProp) <- getCurrentEpochNo epochStateView sbe
+  H.note_ $ "Epoch after first prop: " <> show epochAfterFirstProp
+
+  void $ waitUntilEpoch (File configurationFile) (File socketPath) (EpochNo (epochAfterFirstProp + 2))
+  dRepActivityAfterFirstProp <- getDRepActivityValue execConfig
+
+  dRepActivityAfterFirstProp === 4 -- This is what we just set, should work
+
+  result1 <- checkDRepState sbe (File configurationFile) (File socketPath) execConfig
+                            (Just . map (drepExpiry . snd) . Map.toList)
+  H.note_ $ "DRep expiration dates: " <> show result1
```

Comment:
It depends on when the epoch was registered and because it is not instantaneous, and because epochs are so fast in testnet, there is considerable fluctuation (of several epochs).

Reply:
It depends on which epoch the DRep was registered and because it takes time and epochs are so fast in testnet, there is considerable fluctuation (of several epochs).